splunklogger.js

```diff
@@ -169,16 +250,68 @@ SplunkLogger.prototype._initializeConfig = function(config) {
         ret.protocol = config.protocol || ret.protocol || defaultConfig.protocol;
         ret.level = config.level || ret.level || defaultConfig.level;
 
+        ret.maxRetries = config.maxRetries || ret.maxRetries || defaultConfig.maxRetries;
+        ret.maxRetries = parseInt(ret.maxRetries, 10);
+        if (isNaN(ret.maxRetries)) {
+            throw new Error("Max retries must be a number, found: " + ret.maxRetries);
+        }
+        else if (ret.maxRetries < 0) {
+            throw new Error("Max retries must be a positive number, found: " + ret.maxRetries);
+        }
+
         // Start with the default autoFlush value
         ret.autoFlush = defaultConfig.autoFlush;
         // Then check this.config.autoFlush
         if (this.hasOwnProperty("config") && this.config.hasOwnProperty("autoFlush")) {
-            ret.autoFlush = ret.autoFlush;
+            ret.autoFlush = this.config.autoFlush;
         }
         // Then check the config.autoFlush, the function argument
         if (config.hasOwnProperty("autoFlush")) {
             ret.autoFlush = config.autoFlush;
         }
+        // Convert autoFlush value to boolean
+        ret.autoFlush = !!ret.autoFlush;
+
+        ret.maxBatchCount = config.maxBatchCount || ret.maxBatchCount || defaultConfig.maxBatchCount;
+        ret.maxBatchCount = parseInt(ret.maxBatchCount, 10);
+        if (isNaN(ret.maxBatchCount)) {
+            throw new Error("Max batch count must be a number, found: " + ret.maxBatchCount);
+        }
+        else if (ret.maxBatchCount < 0) {
+            throw new Error("Max batch count must be a positive number, found: " + ret.maxBatchCount);
+        }
+
+        ret.maxBatchSize = config.maxBatchSize || ret.maxBatchSize || defaultConfig.maxBatchSize;
+        ret.maxBatchSize = parseInt(ret.maxBatchSize, 10);
+        if (isNaN(ret.maxBatchSize)) {
```

Comment:
Added lots of validation util functions.

I took a slightly different approach for that scenario by offering recommended defaults. Whereas in Java & .Net we define public const/static values for recommended batching, but they're never used in the constructor.